src/common/procedure/src/store.rs

```diff
@@ -81,14 +81,15 @@ impl ProcedureStore {
             step,
         };
         let key = ParsedKey {
+            prefix: self.proc_path.to_string(),
```

Comment:
I think we don't need to add the prefix as a field of `ParsedKey` (also avoid some allocations). The key itself supports Display
```rust
format!("{}{}", self.proc_path, parsed_key)
```

Or maybe you can refactor proc_path_with and reuse this method
```rust
#[inline]
pub(crate) fn proc_path_with<T: Display>(&self, path: &T) -> String {
    format!("{}{path}/", self.proc_path)
}

let key = self.proc_path_with(&parsed_key);
```

Another benefit is you could avoid `format("{procedure_id}/")`
```rust
self.proc_path_with(&procedure_id);
```

Reply:
> I think we don't need to add the prefix as a field of `ParsedKey` (also avoid some allocations). The key itself supports Display

But the `parse_str` need the prefix.

> ```rust
> format!("{}{}", self.proc_path, parsed_key)
> ```
> 
> Or maybe you can refactor proc_path_with and reuse this method
> 
> ```rust
> #[inline]
> pub(crate) fn proc_path_with<T: Display>(&self, path: &T) -> String {
>     format!("{}{path}/", self.proc_path)
> }
> 
> let key = self.proc_path_with(&parsed_key);
> ```
> 
> Another benefit is you could avoid `format("{procedure_id}/")`
> 
> ```rust
> self.proc_path_with(&procedure_id);
> ```

It's a good idea.

Reply:
It's more convenient to use `key.to_string()`,  so I choose another way to avoid cloning the prefix and add back `proc_path!` macro. PTAL.